packages/common/src/utils/requestValidation.ts

```diff
@@ -0,0 +1,51 @@
+import { NextFunction, Request, Response } from 'express';
+import { ParamsDictionary, RequestHandler } from 'express-serve-static-core';
+import { ZodError, ZodSchema, z } from 'zod';
+import { BadRequest, InternalServerError } from './errors.js';
+
+export declare type RequestValidation<TParams, TQuery, TBody, TFiles> = {
+	params?: ZodSchema<TParams>;
+	query?: ZodSchema<TQuery>;
+	body?: ZodSchema<TBody>;
+	files?: ZodSchema<TFiles>;
+};
+
+/**
+ * Validate the body, path params or files using Zod parse
+ * @param schema Zod objects used to validate request
+ * @returns Throws a Bad Request when validation fails
+ */
+export function validateRequest<TParams = any, TQuery = any, TBody = any, TFiles = any>(
+	schema: RequestValidation<TParams, TQuery, TBody, TFiles>,
+	handler: RequestHandler<ParamsDictionary, any, TBody>,
```

Comment:
Since we are validating params and query, we can also specify those types: `RequestHandler<TParams, any, TBody, TQuery>`

Reply:
Removing Params, Query, files validation as types can not be passed to request